neo/Prompt/Commands/SC.py

```diff
@@ -127,11 +132,91 @@ def command_desc(self):
         p6 = ParameterDesc('returntype', 'the returntype of the smart contract output')
         p7 = ParameterDesc('inputs', 'the test parameters fed to the smart contract, or use "--i" for prompted parameter input')
         p8 = ParameterDesc('--no-parse-addr', 'a flag to turn off address parsing when input into the smart contract', optional=True)
-        p9 = ParameterDesc('--from-addr', 'source address to take fee funds from (if not specified, take first address in wallet)\n\n'
+        p9 = ParameterDesc('--from-addr',
+                           'source address to take fee funds from (if not specified, take first address in wallet)\n\n'
                            f"{' ':>17} Usage Examples:\n"
                            f"{' ':>20} load_run path.py True False False 0710 05 input1 input2\n"
                            f"{' ':>20} load_run path.py True False False 0710 05 --i\n\n"
                            f"{' ':>17} For more information about parameter types see\n"
                            f"{' ':>17} https://neo-python.readthedocs.io/en/latest/data-types.html#contractparametertypes\n", optional=True)
         params = [p1, p2, p3, p4, p5, p6, p7, p8, p9]
         return CommandDesc('load_run', 'load a specified smart contract (.avm) file and test it', params=params)
+
+
+class CommandSCDeploy(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        wallet = PromptData.Wallet
+        if not wallet:
+            print("Please open a wallet")
+            return False
+
+        if len(arguments) < 6:
+            print("Please specify the required parameters")
+            return False
+
+        args, from_addr = PromptUtils.get_from_addr(arguments)
+
+        path = args[0]
+
+        try:
+            needs_storage = bool(util.strtobool(args[1]))
+            needs_dynamic_invoke = bool(util.strtobool(args[2]))
+            is_payable = bool(util.strtobool(args[3]))
+
+        except ValueError:
+            print("Invalid boolean option")
+            return False
+
+        params = args[4]
+        return_type = args[5]
+
+        try:
+            function_code = LoadContract(path, needs_storage, needs_dynamic_invoke, is_payable, params, return_type)
+        except (ValueError, Exception) as e:
+            print(str(e))
+            return False
+
+        contract_script = GatherContractDetails(function_code)
+        if contract_script:

```

Comment:
I think that in general early returns are better than `if ... else return`.
Here, 
```
if not contract_script:
    print("Failed to generate deploy script")
    return False
```

would lead to simpler code (imo)